fileutil/fileutil.go

```diff
@@ -4,13 +4,46 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"github.com/bitrise-io/go-utils/pathutil"
 	"io/ioutil"
 	"log"
 	"os"
-
-	"github.com/bitrise-io/go-utils/pathutil"
+	"path/filepath"
 )
 
+// FileWriter ...
+type FileWriter interface {
+	Write(path string, value string, mode os.FileMode) error
+}
+
+type osFileWriter struct{}
+
+// NewOsFileWriter ...
+func NewOsFileWriter() FileWriter {
```

Comment:
The "os" term just been used as a shitty replacement for "default" or "real" we should come up with a consistent naming for the "real" implementations that doesn't smell like shit :) I mean I'm kinda happier with `defaultFileWriter/realFileWriter` than `OsFileWriter`.

What I would be even more happier with is `FileWriter` and `IFileWriter` for that matter :)

Reply:
If there's no other possible implementation of `FileWriter`, we can simply call this constructor `NewFileWriter` (see my other comment above about naming these kinds of interfaces/functions). And for the implementing `struct`, we can simply call it `fileWriter`.

Reply:
Not sure having fileWriter and FileWriter is wise. Other than that I agree.

Reply:
As discussed above, we settled with `defaultFileWriter`.